support/u-boot/2021.10/patches/0001-Revert-rockchip-Fix-MMC-boot-order.patch

```diff
@@ -0,0 +1,35 @@
+From ba65875dc7311d251f2e4c12179c9a589e997477 Mon Sep 17 00:00:00 2001
+From: Samuel Dionne-Riel <samuel@dionne-riel.com>
+Date: Sun, 6 Feb 2022 20:16:55 -0500
+Subject: [PATCH] Revert "rockchip: Fix MMC boot order"
+
+This reverts commit b212ad24a604b00b240add35516b7381965deb31.
+
+This "fix" goes against the preferred boot order for Tow-Boot, which
+prefers internal storage first.
```

Comment:
There are two orders to consider:

 * Platform Firmware Startup order
 * Operating System Boot order

This is about the latter, the OS boot order.

As long as the platform firmware works, the OS boot order is always trivial-enough to change via user input.

Note that Tow-Boot is usable on many platforms, it is not specific to a particular type of platform, or a specific device.

With all supported platforms, user input is available to change the OS Boot order, even without serial. Though for some (SBCs mainly) it will require a keyboard/display if serial is not available to do so nicely.

The **default** OS boot order in Tow-Boot should always prefer an internal boot source. To boot an external boot source (SD, USB, other), user input must be made. Some would call it a security measure; it is not. It is purely a user experience issue.

Note that I said *default* OS boot order. This is because *down the line*, once things progress to the desired goals, it is intended to provide an interface to select your preferred OS boot order, like you would with a boring old BIOS.

* * *

As for the *platform firmware startup order*, this is a mostly immutable property of the paltform, defined in what is commonly called the "maskrom" code for the SoC. The upstream change, and this revert, change nothing about this. Before/after either change/revert, the situation is the same.

Reply:
Got it, thanks for the detailed answer!

> With all supported platforms, user input is available to change the OS Boot order, even without serial

This is the point I wasn't sure about after a quick look at the existing patches, but after looking into more details (including the phone/tablet PR) it makes sense.